src/overlays/effects/ovl_Effect_Ss_G_Spk/z_eff_ss_g_spk.c

```diff
@@ -1,30 +1,145 @@
 /*
  * File: z_eff_ss_g_spk.c
  * Overlay: ovl_Effect_Ss_G_Spk
- * Description:
+ * Description: Sparks
  */
 
 #include "z_eff_ss_g_spk.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rPrimColorR regs[0]
+#define rPrimColorG regs[1]
+#define rPrimColorB regs[2]
+#define rPrimColorA regs[3]
+#define rEnvColorR regs[4]
+#define rEnvColorG regs[5]
+#define rEnvColorB regs[6]
+#define rEnvColorA regs[7]
+#define rTexIndex regs[8]
+#define rScale regs[9]
+#define rScaleStep regs[10]
 
 #define PARAMS ((EffectSsGSpkInitParams*)initParamsx)
 
-s32 EffectSsGSpk_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
-void func_80978F40(PlayState* play, u32 index, EffectSs* this);
-void func_80979068(PlayState* play, u32 index, EffectSs* this);
+u32 EffectSsGSpk_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+void EffectSsGSpk_Update(PlayState* play, u32 index, EffectSs* this);
+void EffectSsGSpk_UpdateNoAccel(PlayState* play, u32 index, EffectSs* this);
 void EffectSsGSpk_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_G_Spk_InitVars = {
     EFFECT_SS_G_SPK,
     EffectSsGSpk_Init,
 };
 
-#endif
+static TexturePtr sSparkTextures[] = {
+    gElectricSpark1Tex,
+    gElectricSpark2Tex,
+    gElectricSpark3Tex,
+    gElectricSpark4Tex,
+};
```

Comment:
So these are drawn sequentially because the index is incremented?

Might want to add that as a comment that this is meant to be a sort of manual animation

Reply:
And since it's an array, maybe start at 0?

Reply:
I'd prefer to keep the asset names synced with OoT for now which start at 1, though I do see the merit of starting at 0. I can come back to it when all the effects are done/in. Same sentiment with the comment, as it would apply to more then just this effect, so kind of want to wait to see how it goes. (Don't worry I'll add this to my todo list so it won't get lost).